File: 프로그래머스/2/12953. N개의 최소공배수/N개의 최소공배수.py
def gcd(a, b):
    while b != 0:
        a, b = b, a%b
    return a

def solution(arr):
    answer = 0
    
    for i in arr:
        answer = i if answer == 0 else answer * i // gcd(answer, i)
    
    return answer

File: 프로그래머스/2/12953. N개의 최소공배수/N개의 최소공배수.java
class Solution {
    
    public int gcd(int a, int b) {
        if (a%b == 0)
            return b;
        return gcd(b, a%b);
    }
    
    public int solution(int[] arr) {
        int answer = 1;
        for (int num : arr) {
            answer = answer * num / gcd(answer, num);
        }
        return answer;
    }
}